### PR TITLE
Move emqx to apps dir

### PR DESCRIPTION
**Pull request by zmstone**
## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] In case of non-backward compatible changes, reviewer should check this item as a write-off, and add details in **Backward Compatibility** section

## More information

### CODE REVIEW

**Review by terry-xiaoyu — APPROVED** [review]

**Review by Rory-Z — DISMISSED** [review]

File: apps/emqx/rebar.config [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+{deps,
+    [ {gproc, {git, "https://github.com/uwiger/gproc", {tag, "0.8.0"}}}
+    , {jiffy, {git, "https://github.com/emqx/jiffy", {tag, "1.0.5"}}}
+    , {cowboy, {git, "https://github.com/emqx/cowboy", {tag, "2.8.2"}}}
+    , {esockd, {git, "https://github.com/emqx/esockd", {tag, "5.8.0"}}}
+    , {ekka, {git, "https://github.com/emqx/ekka", {tag, "0.8.1"}}}
+    , {gen_rpc, {git, "https://github.com/emqx/gen_rpc", {tag, "2.5.1"}}}
+    , {cuttlefish, {git, "https://github.com/emqx/cuttlefish", {tag, "v3.3.5"}}}
+    , {pbkdf2, {git, "https://github.com/emqx/erlang-pbkdf2.git", {branch, "2.0.4"}}}
+    , {emqtt, {git, "https://github.com/emqx/emqtt", {tag, "1.2.3.1"}}}
```
Comment (x1001100011) [inline apps/emqx/rebar.config:20]: looks like emqtt is only used by the test logic, could we move it to the test deps?
or `mqtt` and `emqtt` app will be cyclically dependent.

File: apps/emqx/test/emqx_cm_SUITE.erl [hunk 2]
```diff
@@ -109,9 +109,13 @@ t_open_session(_) ->
     emqx_cm:unregister_channel(<<"clientid">>),
     ok = meck:unload(emqx_connection).
 
+rand_client_id() ->
+    list_to_binary("client-id-" ++ integer_to_list(erlang:system_time())).
```
Comment (x1001100011) [inline apps/emqx/test/emqx_cm_SUITE.erl:113]: just feel `erlang:unique_integer` may be better here
```suggestion
    list_to_binary("client-id-" ++ integer_to_list(erlang:unique_integer([positive]))).
```

File: rebar.config [hunk 3]
```diff
@@ -33,8 +31,6 @@
 
 {post_hooks,[]}.
 
-{erl_first_files, ["src/emqx_logger.erl", "src/emqx_rule_actions_trans.erl"]}.
```
Comment (x1001100011) [inline rebar.config:36]: do we need to add `emqx` as a dependent of other apps to make sure compiling `emqx` app first?

**Review by x1001100011 — COMMENTED** [review]
I still saw codes like `-include("types.hrl")` in this branch, do we want to change all of them to `-include_lib("emqx/include/types.hrl")`

File: rebar.config (hunk 3 above)
Comment (zmstone) [reply rebar.config:36]: rebar3 should detect this dependency.
for some reason this did not work when src was in root dir.

File: apps/emqx/test/emqx_cm_SUITE.erl (hunk 2 above)
Comment (zmstone) [reply apps/emqx/test/emqx_cm_SUITE.erl:113]: system_time is unique enough for tests.

File: apps/emqx/rebar.config (hunk 1 above)
Comment (zmstone) [reply apps/emqx/rebar.config:20]: thanks. moved it to test profile.

**Comment by zmstone** [comment]
> I'm still seeing codes like `-include("types.hrl")` in this branch, do we want to change all of them to `-include_lib("emqx/include/types.hrl")`

Make sense. will do

**Comment by zmstone** [comment]
> > I'm still seeing codes like `-include("types.hrl")` in this branch, do we want to change all of them to `-include_lib("emqx/include/types.hrl")`
> 
> Make sense. will do

hmm. seem to be all local includes from modules in `app/emqx/src` ?

**Comment by zmstone** [comment]
moved to: https://github.com/emqx/emqx/pull/4925/files

**Pull request closed without merging**